runtime/embed/process.go

```diff
@@ -94,6 +97,44 @@ func process(worker *Worker, loggingBackend sqlq.LogBackend) (shutdown func(time
 						result <- panicWrap(func() error { return fn.Process(jobContext, job) })
 					}()
 
+					// we run a polling checking every 3 seconds whether the job have a
+					// cancelling status if so we change it's status to cancelled.
+					go func() {
+						var poll = func(job *sqlq.Job) (r int64, err error) {
+							var tx *sql.Tx
+							var rows int64
+							if tx, err = worker.db.Begin(); err != nil {
+								return 0, err
+							}
+							defer func() { _ = tx.Rollback() }()
+
+							if rows, err = sqlq.Cancelled(tx, job); err != nil {
+								return 0, err
+							}
+
+							err = tx.Commit()
+							return rows, err
+						}
+
+						//lint:ignore S1000 we want to make sure we use a for loop instead a for range loop
+						for {
+							//lint:ignore S1037 we don't want to sleep the context, instead we want to wait 3 seconds for this to execute
+							select {
+							case <-time.After(3 * time.Second):
+								var rows int64
+								if rows, err = poll(job); err != nil {
+									log.Printf("failed to : %#v", err)
+									return
+								}
+
+								if rows > 0 {
+									<-cancelled
+								}
+
```

Comment:
We would need another `case` for `<-ctx.Done()` to terminate this goroutine after this job is done. Else we'd leak this goroutine.